modules/Vtiger/duplicate.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+error_reporting("E_ALL & ~E_NOTICE & ~E_STRICT & ~E_DEPRECATED & ~E_WARNING");
+require_once "include/utils/utils.php";
+
+if(isset($_REQUEST['module_name']) && isset($_REQUEST['record_id']))
+{
+	$currentModule = vtlib_purify($_REQUEST['module_name']);
+	$record_id = vtlib_purify($_REQUEST['record_id']);
+	
+	require_once 'modules/'.$currentModule.'/'.$currentModule.'.php';
+	$focus = new $currentModule();
+	$focus->retrieve_entity_info($record_id, $currentModule);
+
+	//Duplicate Records that this Record is dependent of 
+	foreach ($focus->column_fields as $fieldname => $value) {
+		$sql = "SELECT 	* FROM vtiger_field WHERE columnname = ? AND uitype IN (10,50,51,57,58,59,73,68,76,75,81,78,80)";
+		$result = $adb->pquery($sql , array($fieldname));
+		if($adb->num_rows($result) == 1 && $value !=0)
+		{
+			$sql = "SELECT setype FROM vtiger_crmentity WHERE crmid = ?";
+			$get_module = $adb->pquery($sql , array($value));
+			$module = $adb->query_result($get_module , 0 , "setype");
+			require_once "modules/" . $module ."/". $module .".php";
+			$entity = new $module();
+			$entity->retrieve_entity_info($value,$module);
+			$entity->save($module);
+			$new_entity_id = $entity->id;
+			$focus->column_fields[$fieldname] = $new_entity_id;
+		}
+	}
+
+	$focus->save($currentModule);
```

Comment:
I think this won't work correctly. If I am correct, when you get information from the database it comes in the database format but when you save it, it has to be in the current user's format, so some fields may be saved incorrectly. You can try this with special characters, date and numeric fields.
http://corebos.org/documentation/doku.php?id=en:devel:saverecordfromwithin

If I am correct you have to pass column_fields through DataTransform::sanitizeRetrieveEntityInfo like this:

https://github.com/tsolucio/corebos/blob/master/modules/com_vtiger_workflow/tasks/VTUpdateFieldsTask.inc#L112